src/modules/System/Service.php

```diff
@@ -379,22 +379,17 @@ public function getEnv($ip)
 
     public function getCurrentUrl()
     {
-        $request = $this->di['request'];
-        $pageURL = 'http';
-        $https = $request->getScheme();
-        if (isset($https) && 'https' == $https) {
-            $pageURL .= 's';
-        }
-        $pageURL .= '://';
+        $pageScheme = $_SERVER['HTTPS'] ? 'https' : 'http';
```

Comment:
This is probably fine, but we may want to consider implementing a helper function somewhere, since from my understanding there is no true "one way" to detect HTTPs.
See: https://www.designcise.com/web/tutorial/how-to-check-for-https-request-in-php

The TLDR: Some web servers may set `$_SERVER['HTTPS']` to `off` or even skip using it entirely.

Reply:
You're right - there isn't an easy easy way to detect valid SSL - and perhaps we should trust config? If so, this could be updated to query config.

At the end of day, there may be people running the system internally using self-signed certs.

Reply:
Or even on a proxy that forward everything to http:// internally 

Probally thrust the config instead and only add an detection in the installer should be fine imo..

Reply:
We have forced HTTPs rewriting as an option (and one I think we should keep), so we already have an internal use for detecting if the user is connected via HTTPs. I think a properly flushed out function to check wouldn't hurt.

For reference: https://github.com/FOSSBilling/FOSSBilling/blob/main/src/load.php#L356